block/produce.go

```diff
@@ -94,48 +114,50 @@ func (m *Manager) produceBlock(ctx context.Context, allowEmpty bool) error {
 		newHeight = height + 1
 		lastCommit, err = m.store.LoadCommit(height)
 		if err != nil {
-			return fmt.Errorf("while loading last commit: %w", err)
+			return nil, nil, fmt.Errorf("load commit: height: %d: %w: %w", height, err, ErrNonRecoverable)
 		}
 		lastBlock, err := m.store.LoadBlock(height)
 		if err != nil {
-			return fmt.Errorf("while loading last block: %w", err)
+			return nil, nil, fmt.Errorf("load block after load commit: height: %d: %w: %w", height, err, ErrNonRecoverable)
 		}
 		lastHeaderHash = lastBlock.Header.Hash()
 	}
 
 	var block *types.Block
+	var commit *types.Commit
 	// Check if there's an already stored block and commit at a newer height
 	// If there is use that instead of creating a new block
-	var commit *types.Commit
 	pendingBlock, err := m.store.LoadBlock(newHeight)
 	if err == nil {
-		m.logger.Info("Using pending block", "height", newHeight)
+		// Using an existing block
+
 		block = pendingBlock
 		commit, err = m.store.LoadCommit(newHeight)
 		if err != nil {
-			m.logger.Error("Loaded block but failed to load commit", "height", newHeight, "error", err)
-			return err
+			return nil, nil, fmt.Errorf("load commit after load block: height: %d: %w: %w", newHeight, err, ErrNonRecoverable)
 		}
+		m.logger.Info("using pending block", "height", newHeight)
 	} else {
```

Comment:
maybe worth better checking the `m.store.LoadBlock` error
to distinguish block not found vs error

Reply:
good spot! done